client/matching/loadbalancer.go

```diff
@@ -88,10 +88,10 @@ func NewLoadBalancer(
 ) LoadBalancer {
 	lb := &defaultLoadBalancer{
 		namespaceIDToName:   namespaceIDToName,
-		nReadPartitions:     dc.GetTaskQueuePartitionsProperty(dynamicconfig.MatchingNumTaskqueueReadPartitions),
-		nWritePartitions:    dc.GetTaskQueuePartitionsProperty(dynamicconfig.MatchingNumTaskqueueWritePartitions),
-		forceReadPartition:  dc.GetIntProperty(dynamicconfig.TestMatchingLBForceReadPartition, -1),
-		forceWritePartition: dc.GetIntProperty(dynamicconfig.TestMatchingLBForceWritePartition, -1),
+		nReadPartitions:     dc.GetIntByTaskQueue(dynamicconfig.MatchingNumTaskqueueReadPartitions),
```

Comment:
How do you feel about reversing things like the way the metrics work with `With(handler)` now? e.g.
```suggestion
		nReadPartitions:     dynamicconfig.MatchingNumTaskqueueReadPartitions.Get(dc),
```
It wouldn't add any safety, and actually the generated code would hardly change. It saves some typing at the callers, which is nice.

(At that point `Collection` hardly has anything in it but a reference to the client and a logger, so we could maybe even try to remove it, but I think I might want to keep it around for subscription management.)

Reply:
There's less typing with the second approach which I like, and it means you can't accidentally provide the dynamic config value to the wrong `GetFooByBar` function

Reply:
On this note: I wish we could specify the dynamic config in the file/module where it's relevant rather than in a single DynamicConfig package. something like
```
var numTaskQueuePartitions = dynamicconfig.NewBLAH(...)
...
    nReadPartitions: numTaskQueuePartitions.Get(dc)
```

is a nicer interface

Reply:
> it means you can't accidentally provide the dynamic config value to the wrong GetFooByBar function

You already can't do that (and have it build) because of the types. But yes, it's probably easier typing-wise to reverse it.

Reply:
> I wish we could specify the dynamic config in the file/module where it's relevant rather than in a single DynamicConfig package

You can totally do that, both before and after this change. (I think ~3 years ago you couldn't.)

Since that's more of an organization thing than a functional thing, how about we discuss that as a team and maybe do it in a follow-up PR if people like it? Or do you think it's better to do all at once?